### PR TITLE
feat: Support Indexing options for Astra DB columns

### PR DESCRIPTION
**Pull request by erichare**
This pull requests adds support for specifying the indexing options for various columns in Astra DB, allowing users to avoid a situation where long text columns are by-default indexed.

### CODE REVIEW

**Comment by erichare** [comment]
Marked it as ready for review now after some testing internally with the team. The primary change here is we give flexibility in which Astra DB fields to index. By default, we deny indexing on the metadata field (which can sometimes be very long due to the parsed HTML from PDFs) but users can override this either in advance, or at collection creation time.

**Comment by potter-potter** [comment]
@erichare I'll check this out tomorrow. Thanks.

**Comment by erichare** [comment]
Thanks @potter-potter !

File: unstructured/ingest/connector/astra.py [hunk 1]
```diff
@@ -67,23 +68,89 @@ def to_dict(self, **kwargs):
...
+                                " on storing long texts. Consider reindexing anew on a"
+                                " fresh collection."
+                            ),
+                            UserWarning,
+                            stacklevel=2,
+                        )
+                        self._astra_db_collection = self._astra_db.collection(
+                            collection_name=collection_name,
+                        )
+                else:
```
Comment (potter-potter) [inline unstructured/ingest/connector/astra.py:150]: This else doesn't seem right. It should run if the try completes successfully.

File: unstructured/ingest/connector/astra.py [hunk 2]
```diff
@@ -35,6 +34,8 @@ class SimpleAstraConfig(BaseConnectorConfig):
     access_config: AstraAccessConfig
     collection_name: str
     embedding_dimension: int
+    namespace: t.Optional[str] = None
```
Comment (potter-potter) [inline unstructured/ingest/connector/astra.py:37]: if you want them exposed to the cli, namespace and requested_indexing_policy need to be added to 

https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/cli/cmds/astra.py

File: unstructured/ingest/connector/astra.py [hunk 3]
```diff
@@ -67,23 +68,89 @@ def to_dict(self, **kwargs):
...
-                options={"indexing": {"deny": NON_INDEXED_FIELDS}},
-            )
+            try:
+                # Create and connect to the newly created collection
+                self._astra_db_collection = self._astra_db.create_collection(
+                    collection_name=collection_name,
+                    dimension=embedding_dimension,
+                    options=_options,
+                )
+            except APIRequestError:
```
[the remark this answers is not in the document]
Comment (erichare) [reply unstructured/ingest/connector/astra.py:101]: @potter-potter you are right. to be honest, this logic was just taken from another integration with a different library, but i think a lot of it is legacy in terms of how the library behaves. In this instance, just creating the collection (which as you said, will connect if the collection already exists) will be enough.

If there is an APIRequestError due to legacy indexing settings, which is what the logic was intending to handle, there are obvious ways now for the user to address it, AND i dont expect this to be the case for Unstructured users in particular.

All that said, good call out and i'll just clean up this code by removing the try / except clause - connect or create, any error will be raised (which should almost never occur)

**Comment by erichare** [comment]
@potter-potter just tried to address your comments. agree with all of them and explained a little for the prior (misguided, lol) motivation :) let me know if this looks better!

File: unstructured/ingest/connector/astra.py (hunk 2 above)
Comment (erichare) [reply unstructured/ingest/connector/astra.py:35]: Done! i tried to match the data model to the "required" option for the CLI, hopefully this looks good.

File: unstructured/ingest/cli/cmds/astra.py [hunk 4]
```diff
@@ -38,10 +38,25 @@ def get_cli_options() -> t.List[click.Option]:
...
+                required=False,
+                default=None,
+                type=str,
+                help="The Astra DB namespace to write into.",
+            ),
+            click.Option(
+                ["--requested-indexing-policy"],
+                required=False,
+                default=None,
+                type=str,
```
Comment (potter-potter) [inline unstructured/ingest/cli/cmds/astra.py:57]: import Dict at the top 
`from unstructured.ingest.cli.interfaces import CliConfig, Dict`

```
type=Dict(),
help="The indexing policy to use for the collection."
'example: \'{"blablabla":"blablabla"}\' ',
```

**Comment by potter-potter** [comment]
@erichare This is looking good. I can take over once you make the little dict change.

File: unstructured/ingest/cli/cmds/astra.py (hunk 4 above)
Comment (erichare) [reply unstructured/ingest/cli/cmds/astra.py]: Updated!

**Comment by erichare** [comment]
> @erichare This is looking good. I can take over once you make the little dict change.

Thanks @potter-potter ! I made the update, does it look okay?

**Comment by potter-potter** [comment]
@erichare Looking good! I'll bring it to the finish line tomorrow. Thanks!

**Comment by erichare** [comment]
> @erichare Looking good! I'll bring it to the finish line tomorrow. Thanks!

Thank you very much!

**Comment by potter-potter** [comment]
@erichare just to keep you updated. I have this in a branch. And was going to just include the `feat: Astra DB Source Connector Support` at the same time. But Astra DB Source has some issues I need to debug. So working on that. Will keep you updated. And may ask for your help.

**Comment by erichare** [comment]
Closing as this has been merged through another PR

**Pull request closed without merging**